### PR TITLE
DBZ-4980 Support CTE with create view statement in MySQL parser

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-4980

### CODE REVIEW

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4 [hunk 1]
```diff
@@ -613,6 +613,7 @@ QUERY:                               'QUERY';
 QUICK:                               'QUICK';
 REBUILD:                             'REBUILD';
 RECOVER:                             'RECOVER';
+RECURSIVE:                           'RECURSIVE';
```
Comment (jpechane) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4:616]: @ani-sha As you are adding a new keyword, can t be used as table/column name? If yes then this should be reflected in the grammart too - `keywordCanBeId`

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4 [hunk 2]
```diff
@@ -241,14 +241,30 @@ createTrigger
       routineBody
     ;
 
+withClause
+    : WITH RECURSIVE? commonTableExpressions
+    ;
+commonTableExpressions
```
Comment (jpechane) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4:247]: Probably an empty line should be inserted

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4 [hunk 3]
```diff
@@ -241,14 +241,30 @@ createTrigger
...
+    : cteName (('(' cteColumnName (',' cteColumnName)* ')')?  AS '(' dmlStatement ')')
+      (',' commonTableExpressions)?
+    ;
+
+cteName
```
Comment (jpechane) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4:252]: Can you please prepare a PR for upstream? I wonder if they would like to have the rules done this way or they'd prefer to use `uid` directly in the rules above. Ideally we should do it in the same way.

**Comment by ani-sha** [comment]
Upstream PR - https://github.com/antlr/grammars-v4/pull/2577

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4 (hunk 3 above)
Comment (ani-sha) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4:253]: @jpechane created the upstream PR. I will wait for any suggestions from them and apply them here also.

**Comment by ani-sha** [comment]
@jpechane Applied changes as per upstream & upstream PR is merged.

**Comment by jpechane** [comment]
@ani-sha Applied, thanks!

**Pull request merged**